File: 240603/특정 수와 근접한 합/sum-close-to-particular-number.java
import java.util.*;
public class Main {
    public static int N, S;
    public static int[] arr;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        S = sc.nextInt();
        arr = new int[N];
        int total = 0;
        for (int i = 0; i < N; i++){
            arr[i] = sc.nextInt();
            total += arr[i];
        }
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < N; i++){
            for (int j = i+1; j <N; j++){
                ans = Math.min(ans, Math.abs(S - (total - arr[i] - arr[j])));
            }
        }
        System.out.println(ans);
    }
}

File: 240613/특정 수와 근접한 합/sum-close-to-particular-number.py
N, S = map(int, input().split())
values = list(map(int, input().split()))

total = sum(values)

ans = int(1e9)
for i in range(N):
    for j in range(i+1, N):
        ans = min(ans,abs(S - (total - values[i] - values[j])))
print(ans)